packages/presentation/lib/screen/login/login_screen.dart

```diff
@@ -60,92 +61,146 @@ class _LoginScreenState extends BlocScreenState<LoginScreen, LoginBloc> {
                 left: Dimens.size12,
               ),
               child: Text(
-                AppLocalizations.of(context)!.profile,
+                SM.current.profile,
                 style: AppTextStyles.sfProSemiBold24px,
               ),
             ),
           ),
           body: Center(
             child: Padding(
               padding: const EdgeInsets.symmetric(horizontal: Dimens.size26),
-              child: Column(
-                mainAxisAlignment: MainAxisAlignment.center,
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: [
-                  Text(
-                    AppLocalizations.of(context)!.userName,
-                    style: AppTextStyles.sfProMediumUnselected12px,
-                  ),
-                  const SizedBox(height: Dimens.size8),
-                  TextField(
-                    controller: bloc.textLoginController,
-                    cursorColor: Colors.white,
-                    style: const TextStyle(
-                      color: AppColorsDark.unselectedColor,
+              child: Form(
+                key: bloc.loginScreenFormKey,
+                autovalidateMode: AutovalidateMode.onUserInteraction,
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      SM.current.userName,
+                      style: AppTextStyles.sfProMediumUnselected12px,
                     ),
-                    decoration: InputDecoration(
-                      prefixIcon: SvgPicture.asset(
-                        ImagesPath.profileImage,
-                        width: Dimens.size18,
-                        height: Dimens.size18,
-                        fit: BoxFit.none,
-                      ),
-                      filled: true,
-                      fillColor: AppColorsDark.secondaryColor,
-                      contentPadding: const EdgeInsets.symmetric(
-                        horizontal: Dimens.size10,
+                    const SizedBox(height: Dimens.size8),
+                    TextFormField(
+                      controller: bloc.textLoginController,
+                      cursorColor: Colors.white,
+                      style: const TextStyle(
+                        color: AppColorsDark.unselectedColor,
                       ),
-                      border: OutlineInputBorder(
-                        borderRadius: BorderRadius.circular(Dimens.size4),
-                        borderSide: const BorderSide(
-                          width: Dimens.size0,
-                          style: BorderStyle.none,
+                      onChanged: (loginText) {
+                        bloc.onChangedLogin(loginText);
```

Comment:
you are using property controller and onChange, select one, there is no point in using both of them

Reply:
disagree
I need to use controller and onChange.

Reply:
it makes no sense to use two, as in our case, one thing can do all the work

- If you will only use the controller, then you can subscribe to the controller to track input.

- If you only use onChange, you can also take the value for the login button from it.

@alexei-kruk-idf Do you agree?

Reply:
@egor-slatvinskiy-idf according to the idea, yes, you can choose one thing, or controller or onChange up to you

Reply:
without (logintext){...} you can write just  bloc.onChangedLogin